src/main/java/in/org/projecteka/hiu/dataflow/DataFlowRequestListener.java

```diff
@@ -42,18 +47,55 @@ public void subscribe() throws ClientError {
         MessageListener messageListener = message -> {
             DataFlowRequest dataFlowRequest = convertToDataFlowRequest(message.getBody());
             logger.info("Received data flow request with consent id : " + dataFlowRequest.getConsent().getId());
-            DataFlowRequest.builder().build();
-            logger.info("Initiating data flow request to consent manager");
-            dataFlowClient.initiateDataFlowRequest(dataFlowRequest)
-                    .flatMap(dataFlowRequestResponse ->
-                            dataFlowRepository.addDataRequest(dataFlowRequestResponse.getTransactionId(), dataFlowRequest))
-                    .block();
+            try{
+                var dataRequestKeyMaterial = dataFlowRequestKeyMaterial();
+                var keyMaterial = keyMaterial(dataRequestKeyMaterial);
+                dataFlowRequest.setKeyMaterial(keyMaterial);
```

Comment:
Instead of mutating the existing object, can you create a new object?

Reply:
Hint: You could use toBuilder

Reply:
Done